### PR TITLE
PTX-17570 Add an optional method to upgrade PX Operator during PX upgrade hops based on the PX endpoint

### PR DESCRIPTION
**Pull request by nikolaypopov**
Add an optional method to upgrade PX Operator during PX upgrade hops based on the PX endpoint

### CODE REVIEW

File: drivers/volume/portworx/portworx.go [hunk 1]
```diff
@@ -2952,6 +2939,107 @@ func (d *portworx) upgradePortworxDaemonset(specGenUrl string) error {
...
+	// Get k8s version
+	kubeVersion, err := d.schedOps.GetKubernetesVersion()
+	if err != nil {
+		return err
+	}
+
+	// Getting PX Operator spec
+	u, err := url.Parse(specGenUrl)
+	if err != nil {
+		fmt.Printf("failed to parse URL [%s], Err: %v", specGenUrl, err)
```
Comment (aghodke1312) [inline drivers/volume/portworx/portworx.go:3002]: log?

File: drivers/volume/portworx/portworx.go [hunk 2]
```diff
@@ -2952,6 +2939,107 @@ func (d *portworx) upgradePortworxDaemonset(specGenUrl string) error {
...
+		var envVars []corev1.EnvVar
+		// Add Release Manifest URL incase of edge spec
+		envVars, err := configurePxReleaseManifestEnvVars(cluster.Spec.Env, specGenUrl)
+		if err != nil {
+			return nil
+		}
+		cluster.Spec.Env = envVars
+		return cluster
+	}
+
```
Comment (aghodke1312) [inline drivers/volume/portworx/portworx.go:2977]: Can you please call stats API here. We are doing something like that in upgrade test. Send stats for upgrade-operator. 
`dash.UpdateStats("px-upgrade-stats", "px-enterprise", "upgrade", majorVersion, statsData)`

File: drivers/volume/portworx/portworx.go (hunk 1 above)
Comment (nikolaypopov) [reply drivers/volume/portworx/portworx.go:3002]: Good catch, fix it in another place I copied this func from

File: drivers/volume/portworx/portworx.go (hunk 2 above)
Comment (nikolaypopov) [reply drivers/volume/portworx/portworx.go:2977]: I don't want to add it inside this function, something like that would need to be added in the test itself

Comment (nikolaypopov) [reply drivers/volume/portworx/portworx.go:2977]: other upgrade tests will probably need that as well, let's do it as a part of a separate PR

**Review by aghodke1312 — APPROVED** [review]

**Review by aghodke1312 — APPROVED** [review]

**Pull request merged**